ASM/c/dungeon_info.c

```diff
@@ -201,39 +204,70 @@ void draw_dungeon_info(z64_disp_buf_t *db) {
 
         // Draw dungeon names
 
-        for (int i = 0; i < dungeon_count; i++) {
-            dungeon_entry_t *d = &(dungeons[i]);
-            int top = start_top + ((icon_size + padding) * i) + 1;
-            text_print_size(d->name, left, top, font_width);
-        }
-        text_flush_size(db, font_width, font_height, 0, 0);
+        if (!SHUFFLE_CHEST_GAME || SHUFFLE_CHEST_GAME == 2) {
```

Comment:
You could merge the “then” and “else” blocks since they're so similar, only checking for `SHUFFLE_CHEST_GAME == 1` in the two places where they should differ, e.g. `i < dungeon_count - (SHUFFLE_CHEST_GAME == 1 ? 0 : 1)`. Same thing for more `if` statements below.